### PR TITLE
tfm: Move modules/tfm together with modules/trusted-firmware-m

### PR DESCRIPTION
**Pull request by joerchan**
Move modules/tfm together with modules/trusted-firmware-m.
The separation between these folders are not really clear to me.
Having two places in the Kconfig menu where TF-M configurations appear are confusing to the user.

### CODE REVIEW

File: subsys/partition_manager/pm.yml.tfm [hunk 1]
```diff
@@ -19,7 +19,7 @@ tfm_secure:
 tfm_nonsecure:
   span: [app]
 
-#ifdef CONFIG_TFM_STORAGE
+#ifdef CONFIG_PM_PARTITION_TFM_STORAGE
```
Comment (SebastianBoe) [inline subsys/partition_manager/pm.yml.tfm:22]: maybe grep for CONFIG_TFM_STORAGE and check if there are any SYSBUILD scripts using it, assuming we are changing it's semantics.

Comment (joerchan) [reply subsys/partition_manager/pm.yml.tfm:22]: I haven't seen anything at all regarding sysbuild and TF-M flash/ram layout.
Our out-of-tree support still rely only on PM support.
@nordicjm Any input here?

Comment (nordicjm) [reply subsys/partition_manager/pm.yml.tfm:22]: Sysbuild uses the existing cmake code for TF-M, you could try the PSA sample building using sysbuild (if it uses this) to check it is still working, or any other sample that does, and ensure it builds and runs properly and that the partitions.yml file has the same sizes and offsets as the child/parent image build

**Review by Vge0rge — COMMENTED** [review]
LGTM

**Review by SebastianBoe — APPROVED** [review]
A much better organization

**Review by Vge0rge — APPROVED** [review]

**Pull request merged**